bees/facebookbee/facebookbee.go

```diff
@@ -76,17 +78,169 @@ func (mod *FacebookBee) Run(eventChan chan bees.Event) {
 	}
 }
 
+func (mod *FacebookBee) handlePermanentPageToken() {
+	if mod.pageAccessToken != "" {
+		// return
+	}
+
+	mod.Logf("Attempting to fetch long lived user access token")
+
+	longToken := mod.fetchLongLivedUserAccessToken()
+
+	if longToken == "" {
+		mod.LogErrorf("No long lived user access token!")
+		return
+	}
+
+	// mod.Logf("Long lived user access token: \"%s\"", longToken)
+	accountID := mod.fetchAccountID(longToken)
+
+	if accountID == "" {
+		mod.LogErrorf("No account id!")
+		return
+	}
+
+	pageToken := mod.fetchPermanentPageAccessToken(accountID, longToken)
+	mod.Logf("pageToken: \"%s\"", pageToken)
+}
+
+func (mod *FacebookBee) fetchLongLivedUserAccessToken() string {
+	// See https://developers.facebook.com/docs/pages/access-tokens/#get-a-long-lived-user-access-token
+	baseURL := "https://graph.facebook.com/oauth/access_token"
+	v := url.Values{}
+	v.Set("grant_type", "fb_exchange_token")
+	v.Set("client_id", mod.clientID)
+	v.Set("client_secret", mod.clientSecret)
+	v.Set("fb_exchange_token", mod.accessToken)
+	graphUrl := baseURL + "?" + v.Encode()
+
+	res, err := http.Get(graphUrl)
+
+	if err != nil || res == nil {
+		mod.LogErrorf("Fetching long lived user access token failed: %v", err)
+		return ""
+	}
+
+	defer res.Body.Close()
+	body, err := ioutil.ReadAll(res.Body)
+
+	if err != nil {
+		mod.LogErrorf("Reading content while fetching long lived user access token failed: %v", err)
+		return ""
+	}
+
+	// mod.Logf("Long lived user access token result: \"%s\"", body)
+
+	type RequestResult struct {
+		AccessToken string `json:"access_token"`
+	}
+
+	var tokenRes RequestResult
+	err = json.Unmarshal(body, &tokenRes)
+
+	if err != nil {
+		mod.LogErrorf("Parsing result while fetching long lived user access token failed: %v", err)
+		return ""
+	}
+
+	return tokenRes.AccessToken
+}
+
+func (mod *FacebookBee) fetchAccountID(accessToken string) string {
+	baseURL := "https://graph.facebook.com/v8.0/me"
+	v := url.Values{}
+	v.Set("access_token", accessToken)
+	v.Set("fields", "id")
+	graphUrl := baseURL + "?" + v.Encode()
+
+	res, err := http.Get(graphUrl)
+
+	if err != nil || res == nil {
+		mod.LogErrorf("Fetching user id failed: %v", err)
+		return ""
+	}
+
+	defer res.Body.Close()
+	body, err := ioutil.ReadAll(res.Body)
+
+	if err != nil {
+		mod.LogErrorf("Reading content while fetching user id failed: %v", err)
+		return ""
+	}
+
+	// mod.Logf("user id result: \"%s\"", body)
+
+	type RequestResult struct {
+		ID string `json:"id"`
+	}
+
+	var tokenRes RequestResult
+	err = json.Unmarshal(body, &tokenRes)
+
+	if err != nil {
+		mod.LogErrorf("Parsing result while fetching user id failed: %v", err)
+		return ""
+	}
+
+	return tokenRes.ID
+}
+
+func (mod *FacebookBee) fetchPermanentPageAccessToken(accountID string, accessToken string) string {
```

Comment:
I would recommend to additionally return any potential errors from this function (instead of just an empty string), so we can cleanly report/log them - at least eventually.